### DISCUSSION TITLE
Specify custom node IDs to `Data`

### DISCUSSION THREAD

**Discussion by harish-vnkt** [pyg-team/pytorch_geometric#5741] [Q&A] [answered]
I have a graph that has specific, alpha-numeric node IDs for each of it's nodes. My edge index defines the connectivity information using these node IDs. I want to create a `Data` object and preserve node ID information for downstream visualization tasks. But the current examples tell me that the node ID is inferred from the feature matrix `x` and that I need a mapping from the custom node IDs to the 0-indexed format used by the `Data` object. Is that interpretation correct? If so, is there a way to define the `Data` object without needing to maintain a mapping?

**Comment by wsad1** [comment, marked as answer]
Your interpretation is correct. The values in `edge_index` are the indices of `x` i.e if you have an edge (1,2) that means `x[1]` is connected to `x[2]`. 
If you don't have a way of computing the alpha-numeric ID from the node id then you'll probably have to maintain a mapping.

**Reply by harish-vnkt** [reply]
Got it. Thank you!
